src/services/MakeStreamZip.php

```diff
@@ -80,7 +80,14 @@ public function __destruct()
      */
     public function getFileName(): string
     {
-        return 'export.elabftw.zip';
+        $this->idArr = explode(' ', $this->idList);
+        if (count($this->idArr) == 1){
+            $this->Entity->setId((int) $this->idArr[0]);
+            $this->Entity->getPermissions();
```

Comment:
Is this the only way to populate the content of the entity? Calls to read() or populate() did not work.

Reply:
In fact I'm currently looking into it because the bin/console dev:populate function is currently broken and the way to populate the object is not great.

As a side note, the code base right now might be a bit messy because I'm heavily working on it and doing big changes so things like that might improve in a close future.

Reply:
Ok thank you for the explanation. I agree, for a developer not too familiar with the code it is right now a bit confusing. But in the process of refactoring all this I assume this is normal :)